cli: add "cilium bpf config list"

mhofstetter opened the pull request:
This PR introduces the possibility to list the content of the BPF map `cilium_runtime_config` via the cilium cli command `cilium bpf config list`.

In addition, these information are dumped during sysdump with the bugtool.

mhofstetter committed:
cli: add "cilium bpf config list"

This commit introduces the possibility to list the content of the BPF
map "cilium_runtime_config" via the cilium cli command `cilium bpf
config list`.

Signed-off-by: Marco Hofstetter <marco.hofstetter@isovalent.com>

mhofstetter committed:
bugtool: dump runtime config map information

This commit introduces that runtime config map related information
are dumped when executing the bugtool / sysdump.

Signed-off-by: Marco Hofstetter <marco.hofstetter@isovalent.com>

mhofstetter commented on pkg/maps/configmap/config_map.go:
```diff
@@ -32,14 +32,21 @@ const (
 	AgentLiveness
```
@dylandreimerink @jrajahalme it looks like the "marker" entry `UsedEntries` is unused and lost its meaning. When L2 Announcement has been introduced, a new entry has been added to the end (whereas the intention of the marker entry was to be the last entry).

should we delete the entry and move `AgentLiveness` to second position?

dylandreimerink replied:
Moving `AgentLiveness` is alright with me, as long as we do it before the feature freeze. Moving after that could pose upgrade issues if the map is not recreated. I also have no idea what `UsedEntries` was supposed to be used for.

mhofstetter replied:
@dylandreimerink thanks - removed the constant (in agent & `common.h`)

mhofstetter committed:
config map: remove constant UsedEntries

This commit removes the unused constant UsedEntries in the runtime
config map. Therefore, AgentLiveness gets moved to position 1.

Signed-off-by: Marco Hofstetter <marco.hofstetter@isovalent.com>

asauber reviewed (approved):
Looks great. The de-duplication of the index strings is nice
